plugins/ekuiper/json_rw.c

```diff
@@ -281,3 +283,217 @@ int json_encode_read_resp(void *json_object, void *param)
 
     return ret;
 }
+
+int json_decode_write_req(char *buf, size_t len, json_write_req_t **result)
+{
+    int               ret      = 0;
+    void *            json_obj = NULL;
+    json_write_req_t *req      = calloc(1, sizeof(json_write_req_t));
+    if (req == NULL) {
+        return -1;
+    }
+
+    json_obj = neu_json_decode_newb(buf, len);
+    if (NULL == json_obj) {
+        goto decode_fail;
+    }
+
+    neu_json_elem_t req_elems[] = {
+        {
+            .name = "node_name",
+            .t    = NEU_JSON_STR,
+        },
+        {
+            .name = "group_name",
+            .t    = NEU_JSON_STR,
+        },
+        {
+            .name = "tag_name",
+            .t    = NEU_JSON_STR,
+        },
+        {
+            .name = "value",
+            .t    = NEU_JSON_VALUE,
+        },
+    };
+    ret = neu_json_decode_by_json(json_obj, NEU_JSON_ELEM_SIZE(req_elems),
+                                  req_elems);
+    if (ret != 0) {
+        goto decode_fail;
+    }
+
+    req->node_name  = req_elems[0].v.val_str;
+    req->group_name = req_elems[1].v.val_str;
+    req->tag_name   = req_elems[2].v.val_str;
+    req->t          = req_elems[3].t;
+    req->value      = req_elems[3].v;
+
+    *result = req;
+    goto decode_exit;
+
+decode_fail:
+    if (req != NULL) {
+        free(req);
+    }
+    ret = -1;
+
+decode_exit:
+    if (json_obj != NULL) {
+        neu_json_decode_free(json_obj);
+    }
+    return ret;
+}
+
+void json_decode_write_req_free(json_write_req_t *req)
+{
+    if (NULL == req) {
+        return;
+    }
```

Comment:
assert

Reply:
This is to keep the semantics with the C `free` function, which does nothing when freeing a NULL pointer.